Clean up DRET text (#1044)
riscv/riscv-cheri: src/cheri/debug-integration.adoc

Reviewer: 1) Fix of punctuation (extra `,`).

2) Attempt to simplify the wording (no change in meaning).

```suggestion
When the hart is in debug mode, the _RISC-V Debug Specification_ does not specify
how the PC is updated, and says that PC-relative instructions may be illegal. This concept
```

Reviewer: There seems to be a leftover piece of text (that should have been deleted):

```suggestion
When debug mode is exited, the hart's <<ddc>> is updated to
the capability stored in <<dddc>>. A debugger may write <<dddc>> to change the
hart's context.
```

On debug mode entry, <<dpc_y>> is updated with the

> capability in <<pcc>> whose address field is set to the address of the next

Author: ```suggestion
```

capability in <<pcc>> whose address field is set to the address of the next

> instruction to be executed upon debug mode exit as described in the _RISC-V Debug Specification_.

Reviewer: Wording suggestion to avoid possible confusion: `is set to` --> `equals to`.

"Is set to" could lead the reader to think that the address field gets somehow changed, which is not the case.

```suggestion
On debug mode entry, <<dpc_y>> is updated with the
capability in <<pcc>>, whose address field equals to the address of the next
instruction to be executed upon debug mode exit as described in the _RISC-V Debug Specification_.
```

When leaving debug mode, the value in <<dpc_y>> is unsealed if it is a <<sentry_cap>> and is written to <<pcc>>.

NOTE: A debugger may write <<dpc_y>> to change where the hart resumes and its mode, permissions, sealing or bounds.
 The value saved from the <<pcc>> on debug mode entry cannot be sealed.

The legalization of <<dpc_y>> follows the same rules described for <<mepc_y>>.

> One method of resuming execution is <<DRET_CHERI>>.

Reviewer: 1) Two terms (synonyms) are used in this section: "debug mode exit" and "resuming execution". If you don't mind, I'd keep using just one of them.

2) I would like to stress also here that the use of DRET is purely an implementation choice. Something along these lines:

```suggestion
One possible implementation choice for debug mode exit is DRET.
```

or

```suggestion
Exiting debug mode can be implemented using DRET, but the implementation is free to choose any other method instead.
```

(or similar)
